ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -7487,6 +7487,83 @@ void HqlCppTranslator::doBuildAssignExecuteWhen(BuildCtx & ctx, const CHqlBoundT
     }
 }
 
+void HqlCppTranslator::doBuildExprPow(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    assertex(expr->numChildren() == 2);
+    
+    IHqlExpression * left = expr->queryChild(0);
+    IHqlExpression * right = expr->queryChild(1);
+
+    bool notIntegerPower = !right->queryType()->isInteger();
+    if (notIntegerPower)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IValue * value = right->queryValue();
+    bool notConstant = (value == nullptr);
+    if (notConstant) 
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    const uint MAX_EXP_INLINE_VALUE = 3;
+    int64_t rightIntVal = value->getIntValue();
+    if (abs(rightIntVal) > MAX_EXP_INLINE_VALUE)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IHqlExpression* outExpr = nullptr;
```

Comment:
A couple of issues:
a. Expression link counts.  The case right==0 will create an expression with a link count of 1, the case right=1 is no linking the result.  The expression is not released - so it will leak for right==0 (and right != 1)
You should check changes carefully for memory leaks (can use valgrind, but need to use the --leakcheck option to ensure everything is cleaned up correctly.

b. Different kinds of expressions.  The same structures are used to represent ECL expressions and c++ expressions, but you need to be very careful about not mixing them.  So buildExpr() expects an ECL expression, but on line 7563 you are passing it a c++ expression.  That could cause problems.

I think in this case you should not be calling buildExpr at all, but should be setting tgt.expr.